### PR TITLE
Take into account empty lines when extracting text content from the appearance

### LINKED ISSUE #17492: Extra line breaks are missing when PDF is loaded
Link to PDF file here:
[document (19).pdf](https://github.com/mozilla/pdf.js/files/13894933/document.19.pdf)

It is reproducible in https://mozilla.github.io/pdf.js/web/viewer.html
Configuration:
- Web browser and its version: Firefox 121.0.1 (64-bit)
- Operating system and its version: Windows 11 Pro 10.0.22631
- PDF.js version: Latest (https://mozilla.github.io/pdf.js/web/viewer.html)
- Is a browser extension: No

Steps to reproduce the problem:
1. Load PDF file
Actual result:
There is no missing blank line in the field between Several and Other
image

Expected result:
If you open PDF in any other PDF viewer (Foxit desktop for example), you will see that empty line is there
image

It is always reproducible with any file and any field.
You can add extra line breaks right in pdf.js, save the file, and it will be gone on the file load.

### PR DESCRIPTION
**Pull request by calixteman**
Fixes #17492.

### CODE REVIEW

File: test/unit/api_spec.js [hunk 1]
```diff
@@ -2550,6 +2550,20 @@ describe("api", function () {
       await loadingTask.destroy();
     });
 
+    it("read content from multline textfield containing an empty line", async function () {
```
Comment (Snuffleupagus) [inline test/unit/api_spec.js:2553]:
```suggestion
    it("read content from multiline textfield containing an empty line", async function () {
```

**Review by Snuffleupagus — APPROVED** [review]
That's even simpler than I imagined :-)

r=me, with all tests passing; thank you!

**Comment by calixteman** [comment]
The failure with `bug1219400.pdf` is an improvement (there's an empty line in the FreeText annotation).

**Pull request merged**